packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -563,22 +564,39 @@ class PageTransitionsTheme with Diagnosticable {
   /// Constructs an object that selects a transition based on the platform.
   ///
   /// By default the list of builders is: [FadeUpwardsPageTransitionsBuilder]
-  /// for [TargetPlatform.android], and [CupertinoPageTransitionsBuilder] for
-  /// [TargetPlatform.iOS] and [TargetPlatform.macOS].
-  const PageTransitionsTheme({ Map<TargetPlatform, PageTransitionsBuilder> builders = _defaultBuilders }) : _builders = builders;
+  /// for [TargetPlatform.android] and [TargetPlatform.fuchsia],
+  /// [CupertinoPageTransitionsBuilder] for [TargetPlatform.iOS], and no
+  /// animated transition for other platforms.
+  ///
+  /// For web apps, there are no default transitions specified, but the app
+  /// can override this by creating a theme with whatever [webBuilders] they
+  /// would like.
+  const PageTransitionsTheme({
+    Map<TargetPlatform, PageTransitionsBuilder> builders = _defaultBuilders,
+    Map<TargetPlatform, PageTransitionsBuilder> webBuilders = _defaultWebBuilders,
```

Comment:
This API feels weird to me, when you build for the web you usually don't care about the native platforms. You probably care about browsers, but that is not the case for the flutter.

I wonder whether we can change this API so that there is just one builder for the web, and then makes the API easier to expand to differentiate each platform if customers really want to do that.

Reply:
> I wonder whether we can change this API so that there is just one builder for the web, and then makes the API easier to expand to differentiate each platform if customers really want to do that.

I did consider that, but we wouldn't be able to add it to the existing `builders` field as there isn't a `TargetPlatform.web`. So if we have to add another field for the web, why not provide a way to specialize it for the underlying host platform? That way if you are writing a PWA and want it to be as native looking as possible you could provide the native transitions for each platform with the `webBuilders` field.

Reply:
Not sure whether this is a good idea, just want to throw options around. Can we do this as a bug fix without adding new API? for example we conditionally sets the _builders to _defaultWebBuilders if kIsWeb is true. This way, the behavior is the same, and user can still customize using the old api by check the kIsWeb themselves. We don't risk exposing a new API that we aren't sure whether it can be useful.

Reply:
That sounds good to me. However, would probably be good to get some inout from the web team around whether defaulting to no transition is the right thing (I think it is).